modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/coordination/hazelcast/HazelcastAgent.java

```diff
@@ -506,4 +510,511 @@ public boolean isActive() {
+            HashmapStringTreeSetWrapper wrapper = this.slotAssignmentMap.get(nodeId);
+            if (wrapper == null) {
+                wrapper = new HashmapStringTreeSetWrapper();
+                queueToSlotMap = new HashMap<String, TreeSet<Slot>>();
+                wrapper.setStringListHashMap(queueToSlotMap);
+                this.slotAssignmentMap.putIfAbsent(nodeId, wrapper);
+            }
+            wrapper = this.slotAssignmentMap.get(nodeId);
+            queueToSlotMap = wrapper.getStringListHashMap();
+            currentSlotList = queueToSlotMap.get(queueName);
+            if (currentSlotList == null) {
+                currentSlotList = new TreeSet<Slot>();
+            }
+
+            //update slot state
+            if (allocatedSlot.addState(SlotState.ASSIGNED)) {
+                //remove any similar slot from hazelcast and add the updated one
+                currentSlotList.remove(allocatedSlot);
+                currentSlotList.add(allocatedSlot);
+                queueToSlotMap.put(queueName, currentSlotList);
+                wrapper.setStringListHashMap(queueToSlotMap);
+                this.slotAssignmentMap.set(nodeId, wrapper);
+            }
+        } catch (HazelcastInstanceNotActiveException ex) {
+            if (AndesKernelBoot.isKernelShuttingDown()) {
+                throw new AndesException("Failed to update slot assignment for queue : " +
+                        queueName + " from node " + nodeId, ex);
+            }
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long getQueueToLastAssignedId(String queueName) throws AndesException {
+        long lastAssignedId = this.lastAssignedIDMap.get(queueName) != null ?
+                this.lastAssignedIDMap.get(queueName) : 0L;
+        return lastAssignedId;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setQueueToLastAssignedId(String queueName, long lastAssignedId) throws AndesException {
+        this.lastAssignedIDMap.set(queueName, lastAssignedId);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Long getNodeToLastPublishedId(String nodeId) throws AndesException {
+        return this.lastPublishedIDMap.get(nodeId);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setNodeToLastPublishedId(String nodeId, long lastPublishedId) throws AndesException {
+        this.lastPublishedIDMap.set(nodeId, lastPublishedId);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TreeSet<String> getMessagePublishedNodes() throws AndesException {
+        TreeSet<String> messagePublishedNodes = new TreeSet<String>();
+        messagePublishedNodes.addAll(this.lastPublishedIDMap.keySet());
+        return messagePublishedNodes;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setSlotState(long startMessageId, long endMessageId, SlotState slotState) throws AndesException {
+
```

Comment:
In hazelcast slot state is already stored in slot object. This method is used only in database slot management.

Added the reason as a comment in PR https://github.com/wso2/andes/pull/276/files